theano/tensor/nnet/abstract_conv.py

```diff
@@ -286,6 +294,9 @@ def get_conv_gradinputs_shape(kernel_shape, top_shape,
 
     if filter_dilation is None:
         filter_dilation = np.ones(len(subsample), dtype='int')
+    if num_groups > 1:
+        assert len(subsample) == 2
+        nkern = nkern * num_groups
```

Comment:
The only way the shapes are different from normal comvolutions are that 
kshp[1] = ishp[1] // num_groups as opposed to kshp[1] = ishp[1] in normal convolutions.

In the forward pass, The top shape isn't influenced by kshp[1], instead its shape would be 
(ishp[0], khsp[0], oh, ow).This is the same of normal convolutions as well, so it does not change.

In the backward pass with respect to weights, in normal convolutions  it would have been
( tshp[1], ishp[1], kh, kw) but for grouped convolution kshp[1] would be ishp[1] // numgroups.

It is similar wrt input as well only that ishp[1] would be kshp[1]  multiplied by num_groups